src/jit/gcencode.cpp

```diff
@@ -4204,9 +4221,26 @@ void GCInfo::gcMakeRegPtrTable(GcInfoEncoder* gcInfoEncoder,
         {
             if (gcCallDescList != nullptr)
             {
-                for (CallDsc* call = gcCallDescList; call != nullptr; call = call->cdNext)
+#if defined(_TARGET_AMD64_) && defined(FEATURE_CORECLR)
+                if (fastPath)
                 {
-                    numCallSites++;
+                    // We have the call count from the previous run.
+                    numCallSites = *callCntRef;
+
+                    // If there are no calls, tell the world and bail.
+                    if (numCallSites == 0)
+                    {
+                        gcInfoEncoderWithLog->DefineCallSites(nullptr, nullptr, 0);
+                        return;
+                    }
```

Comment:
It's unfortunate and undesirable to add a "return" in the middle of this enormous method. Isn't there a way to restructure the code to avoid this, and let it fall through (without adding too much complexity in the way of additional nesting, etc.)?